File: Codes in C++/Dynamic Programming/Palindromic Substrings.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int res=0;
        for(int i=0;i<s.size();i++){
            res+=solve(s,i,i);
            res+=solve(s,i,i+1);
        }
        return res;
}
    int solve(string s, int l, int r){
        int res=0;
        while (l >= 0 && r < s.length() && s[l] == s[r]) {
                res++;
                l--;
                r++;
            }
            return res;
    }
};

File: Codes in Python3/Dynamic Programming/Palindromic Substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        psCount=[0]

        def expand_substring(l,r):
            while l>=0 and r<len(s) and s[l]==s[r]:
                psCount[0]+=1
                l-=1
                r+=1
        for i in range(len(s)):
            expand_substring(i,i)
            expand_substring(i,i+1)

        return psCount[0]
# TC=O(n^2)
# SC=O(1)
# class Solution:
#     def countSubstrings(self, s: str) -> int:
#         psCount=0
#         for i in range(len(s)):
#             #Odd length string
#             l,r=i,i
#             while l>=0 and r<len(s) and s[l]==s[r]:
#                 psCount+=1
#                 l-=1
#                 r+=1
#             l,r=i,i+1
#             while l>=0 and r<len(s) and s[l]==s[r]:
#                 psCount+=1
#                 l-=1
#                 r+=1

#         return psCount